### ISSUE TITLE
Hot-reloading not working on clean install

### ISSUE THREAD

**Issue by JCB-K** [mxstbr/react-boilerplate#255]
Hot-reloading does not seem to be working for me, on a completely fresh install of v3.

Steps to recreate:

```
git clone git@github.com:mxstbr/react-boilerplate
cd react-boilerplate
git checkout v3.0.0
npm run setup
npm run clean
npm start
```

- Open app in Chrome
- Change a string in app/components/HomePage/index.js

Expected behaviour:

string gets changed in browser

Result:

```
[HMR] Waiting for update signal from WDS...
[WDS] Hot Module Replacement enabled.
[WDS] App updated. Recompiling...
[WDS] App hot update...
[HMR] Checking for updates on the server...
XHR finished loading: GET "http://localhost:3000/39a7e2268207762c02b4.hot-update.json".
[HMR] The following modules couldn't be hot updated: (They would need a full reload!)
[HMR]  - 643
[HMR] Nothing hot updated.
[HMR] App is up to date.
```

I'm not entirely sure how to debug this, but I'd be happy to assist.

**Comment by mxstbr** [comment]
Sorry for the late reply, that definitely shouldn't be happening. Is HMR working with the example app instead of the clean slate?

**Comment by JCB-K** [comment]
No problems. I couldn't actually get the example app to work. Will post a
trace when I'm near my computer, but it complained about missing Node
packages (even though they were installed). After removing the example it
works though, except for hot reloading.

**Comment by mxstbr** [comment]
That's very weird. Would love some logs to dig into this!

**Comment by JCB-K** [comment]
Here's a gist with the error from the default example app. As you can see, whatwg-fetch seems to be missing. But just to be clear, this error disappears when removing the example app, then the apps runs without errors. https://gist.github.com/JCB-K/d2bf2695bac0884b5e42715ff7b59462

**Comment by mxstbr** [comment]
Can you try `npm install whatwg-fetch` and then running the example app, and see if that fixes that issue?

**Comment by oliverturner** [comment]
Can confirm that HMR is broken for the example app (having installed `whatwg-fetch`)

**Comment by mxstbr** [comment]
Thanks, closed in favor of #220, which now has quite a priority!

**Comment by JCB-K** [comment]
Yeah same, but the difference is that in the example app nothing is logged in the browser, while the bare boilerplate logs what I pasted above (```The following modules couldn't be hot updated:```). Not sure if that's relevant?

**Comment by oliverturner** [comment]
@JCB-K Just pushed an update... can you pull and feed back on whether this is sorted for you?

**Comment by JCB-K** [comment]
@oliverturner It fixes the example app and hot-reloading works on it as well, but it doesn't work on the barebones app.

```
[HMR] connected
client.js:123 [HMR] bundle rebuilding
client.js:126 [HMR] bundle rebuilt in 1378ms
process-update.js:27 [HMR] Checking for updates on the server...
bootstrap 404c829…:48 XHR finished loading: GET "http://localhost:3000/404c829c712d533fbb00.hot-update.json".hotDownloadManifest @ bootstrap 404c829…:48hotCheck @ bootstrap 404c829…:256check @ process-update.js:64module.exports @ process-update.js:28processMessage @ client.js:139handleMessage @ client.js:65
process-update.js:81 [HMR] The following modules couldn't be hot updated: (Full reload needed)
This is usually because the modules which have changed (and their parents) do not know how to hot reload themselves. See http://webpack.github.io/docs/hot-module-replacement-with-webpack.html for more details.
process-update.js:89 [HMR]  - ./app/components/NotFoundPage/index.js
```

**Comment by oliverturner** [comment]
This is a simple instance of a limitation of HMR via `babel-preset-react-hmre`: functional components can't be hot reloaded.

see the discussion on https://github.com/reactjs/redux/pull/1455 for where we're going to fix this.

**Comment by JCB-K** [comment]
Gotcha. Thanks for the help.

**Comment by hossein761** [comment]
Hi, I am having the same issue here. but the browser does not get refreshed. I see the code recompiles but not the browser. Any ideas?

**Comment by lvnr** [comment]
Same here - webpack rebuilds, the browser won't refresh.

**Comment by dholbrook86** [comment]
Just did a fresh install (twice), updated node and npm to be latest version, ran the example app and it's not hot reloading. Getting the following modules could not be updated error.

**Comment by gihrig** [comment]
This is, unfortunately a known issue:

Hot module reloading does not work with ['function'](https://github.com/mxstbr/react-boilerplate/issues/452#issuecomment-223280275) or ['pureComponent'](https://github.com/mxstbr/react-boilerplate/issues/1220#issuecomment-260356092) based components.

If hot reloading is really important to you, I suggest as a workaround, that you use `React.Component`, even just temporarily, while you are working on page content.

The proposed fix for this issue is being discussed at #1068

**Comment by dholbrook86** [comment]
Thanks for the quick reply. I was pulling my hair out and I even had some friends pull it down to try and get it working. Some said they were able to get HMR working despite the known bug.
